File: JetpacReloaded/Typescript version/WebGLCanvas/ts/tilenoanimation.ts
import { Rectangle } from "../ts_lib/rectangle.js";
import { Tile, TileCollisionType } from "./tile.js";

export class TileNoAnimation extends Tile {

    constructor(value: number, x: number, y: number, tileCollisionType: TileCollisionType) {
        super(value, x, y, tileCollisionType);
    }

    public get TileTexture(): Rectangle {
        return new Rectangle(this.TextureX, this.TextureY, this.m_width, this.m_height);
    }
}

File: JetpacReloaded/C# version/JetpacReloaded/TileNoAnimation.cs
using Microsoft.Xna.Framework;

namespace JetPacReloaded
{
    public class TileNoAnimation : Tile
    {
        public TileNoAnimation(int value, int x, int y, TileCollisionType tileCollisionType) : base(value, x, y, tileCollisionType)
        {
        }

        public override Rectangle TileTexture { get { return new Rectangle(TextureX, TextureY, m_width, m_height); } }
    }
}
